### PR TITLE
Ed25519

### PR DESCRIPTION
**Pull request by arturkow2000**
No known Rust crate can generate RSA on cortex-m in a reasonable time
and hardware crypto (CryptoCell) is not usable currently, mainly due to
closed source driver.

### CODE REVIEW

File: src/client/signing.rs [hunk 1]
```diff
@@ -50,6 +50,10 @@ where
                 }
             }
         }
+        crypto::Key::Ed25519 { .. } => {
+            error!("edd25519 signed objects are not supported");
```
Comment (krystian-hebel) [inline src/client/signing.rs]:
```suggestion
            error!("Ed25519 signed objects are not supported");
```

File: src/client/tpm/mod.rs [hunk 2]
```diff
@@ -164,6 +164,10 @@ where
 
             Ok((secret_copy, id_object, encrypted_secret))
         }
+        crate::certmgr::Key::Ed25519 { .. } => {
+            error!("Credential Activation with ed25519 is not supported");
```
Comment (krystian-hebel) [inline src/client/tpm/mod.rs]:
```suggestion
            error!("Credential Activation with Ed25519 is not supported");
```

File: src/client/signing.rs (hunk 1 above)
Comment (arturkow2000) [reply src/client/signing.rs]: Fixed

File: src/client/tpm/mod.rs (hunk 2 above)
Comment (arturkow2000) [reply src/client/tpm/mod.rs]: Fixed

File: pal/pal_nrf/src/trussed/store.rs [hunk 3]
```diff
@@ -13,10 +13,12 @@ const_ram_storage!(VolatileStorage, 4096);
 // bytes.
 const_ram_storage!(ExternalStorage, 1024);
 
-/// How much storage do we need. For now let's use 64 KiB, may be extended in
-/// future (must be kept in sync with storage size declared in link.x file). Must
-/// be multiple of 4096.
-const PERSISTENT_STORAGE_SIZE: usize = 65536;
+/// How much storage do we need. 64 KiB turned out to be far too less. LittleFS
```
Comment (krystian-hebel) [inline pal/pal_nrf/src/trussed/store.rs]:
```suggestion
/// How much storage do we need. 64 KiB turned out to be far too little. LittleFS
```

File: pal/pal_nrf/src/trussed/store.rs [hunk 4]
```diff
@@ -13,10 +13,12 @@ const_ram_storage!(VolatileStorage, 4096);
...
+/// wastes large amount of space for directory - each empty directory reserves
+/// 2 blocks (4 KiB), see https://github.com/littlefs-project/littlefs/issues/645
```
Comment (krystian-hebel) [inline pal/pal_nrf/src/trussed/store.rs]:
```suggestion
/// 2 blocks (4 KiB each), see https://github.com/littlefs-project/littlefs/issues/645
```

File: pal/pal_nrf/src/trussed/store.rs (hunk 3 above)
Comment (arturkow2000) [reply pal/pal_nrf/src/trussed/store.rs]: Fixed

File: pal/pal_nrf/src/trussed/store.rs (hunk 4 above)
Comment (arturkow2000) [reply pal/pal_nrf/src/trussed/store.rs]: Fixed

**Review by krystian-hebel — APPROVED** [review]

**Pull request merged**